plan/validator.go

```diff
@@ -256,3 +267,29 @@ func (v *validator) checkFieldList(x *ast.FieldList) {
 		}
 	}
 }
+
+func (v *validator) checkCreateTableGrammar(stmt *ast.CreateTableStmt) {
+	for _, colDef := range stmt.Cols {
+		if colDef.Tp.Tp == mysql.TypeString {
+			if colDef.Tp.Flen != types.UnspecifiedLength && colDef.Tp.Flen > 255 {
+				v.err = errors.Errorf("Column length too big for column '%s' (max = 255); use BLOB or TEXT instead", colDef.Name.Name.O)
+				return
+			}
+		}
+	}
+}
+
+func (v *validator) checkCreateIndexGrammar(stmt *ast.CreateIndexStmt) {
+	for i := 0; i < len(stmt.IndexColNames); i++ {
+		name1 := stmt.IndexColNames[i].Column.Name
+		for j := 0; j < len(stmt.IndexColNames); j++ {
```

Comment:
is that necessary?  
I guess columns should be rather small, say, n<5 or n<10, then we don't need care the O(n^2)  avoid some allocations is also a bonus.